R/compare.R

```diff
@@ -176,6 +157,44 @@ order_proxy <- function(proxy, direction = "asc", na_value = "largest") {
   }
 }
 
+# Order proxy ------------------------------------------------------------------
+
+#' Ordering proxy
+#'
+#' Returns a proxy object for ordering. For vctrs_vctr objects, this determines
+#' the behavior of `order()` and `sort()` (via `xtfrm()`). It also powers
+#' `vec_order()` and `vec_sort()` for other vector types.
+#'
+#' The default is to call [vec_proxy_compare()].
+#'
+#' Lists are special cased, and return an integer vector that can be used to
+#' order the elements in the list by first appearance.
+#'
+#' @param x A vector x.
+#' @inheritParams ellipsis::dots_empty
+#' @return A 1d atomic vector or a data frame.
```

Comment:
Should `vec_proxy_order()` always return an integer vector or data frame of integer vectors?

Reply:
I don't think so, because fractional doubles and character vectors can still be ordered as is